test/index.test.js

```diff
@@ -132,5 +134,35 @@ describe('Compat - index.js', function () {
       });
     });
   });
-});
+  describe('hoistable functions', function () {
+    it('should work with defined hoistable function', function (done) {
+      function modify(obj) {
+        obj.anotherKey = 'what';
+        return obj;
+      }
```

Comment:
this actually brings up a good point. Do you need an escape hatch to declare a particular function as not hoistable? e.g. this `modify` function would get async-wrapped assuming that it had a callback, but it doesn't and the async version would never complete, right?

Reply:
I suspect you're going to need to pass an options object to `AwaitWrap` to be able to either pass an allowList or a blockList of things it should touch/not-touch.

Reply:
I see what you mean, we need to support both sync/async case